fe/fe-core/src/main/java/com/starrocks/catalog/ColocateTableIndex.java

```diff
@@ -771,14 +771,78 @@ public long saveColocateTableIndex(DataOutputStream dos, long checksum) throws I
         return checksum;
     }
 
+    public List<GroupId> getOtherGroupsWithSameOrigName(String origName, long dbId) {
+        List<GroupId> groupIds = new ArrayList<>();
+        try {
+            readLock();
+            for (String key : groupName2Id.keySet()) {
+                if (key.split("_")[1].equals(origName) && groupName2Id.get(key).dbId != dbId) {
+                    groupIds.add(groupName2Id.get(key));
+                }
+            }
+        } finally {
+            readUnlock();
+        }
+
+        return groupIds;
+    }
+
+    public GroupId checkColocateSchemaWithGroupInOtherDb(String toCreateGroupName, long dbId,
+                                                      OlapTable toCreateTable) throws DdlException {
+        try {
+            readLock();
```

Comment:
better put it out of try block

Reply:
Actually i have discussed this issue with airbnb guys before, the conclusion is that, it's better to put it into the try block to prevent lock in unlocked state after use, because the readLock() implementation is not alway a single `lock.readLock().lock()` call, there may have some other logic after it, like our db.readLock() implementation, if those logic throws an exception, the unlock action in finally clause won't happend, the lock is locked forever, the former usage of `db.readLock() try {...` is not a best practice considering our current `readLock()/writeLock()` implementation.

Reply:
and besides, `lock.readLock().lock()` is just an atomic var operation, it won't throw exception, so we don't need to worry that we may unlock a not locked lock in finally clause.